### PR TITLE
[image-picker][android] Fix background bug

### LINKED ISSUE #12102: Image picker not working when app with background location is killed
### Summary

On a Doogee android 9 device running background location with foreground service, after using the image picker, when the app is killed then restarted, and then another attempt is made to choose an image, the image picker no longer returns an image from launchCameraAsync.

This error is seen in the logs:
2021-03-02 21:28:40.130 839-850/? W/ActivityManager: startActivity called from non-Activity context; forcing Intent.FLAG_ACTIVITY_NEW_TASK for: Intent { act=android.media.action.IMAGE_CAPTURE flg=0x800003 cmp=com.android.camera/.sdk.activity.CaptureActivity clip={text/uri-list U:content://com.repro.ImagePickerFileProvider/cached_expo_files/ImagePicker/4edea31d-fdcd-4be9-8bcc-93cec713d3ee.jpg} (has extras) }

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

40.0.0

### Environment

 Expo CLI 4.2.1 environment info:
    System:
      OS: macOS 11.2.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 12.18.3 - /usr/local/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 6.14.8 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
      Android SDK:
        API Levels: 28, 29, 30
        Build Tools: 28.0.3, 29.0.2, 30.0.2
        System Images: android-24 | Google APIs Intel x86 Atom, android-27 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.7042882
      Xcode: 12.4/12D4e - /usr/bin/xcodebuild
    npmPackages:
      expo: ~40.0.0 => 40.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: ~0.63.4 => 0.63.4 
      react-native-web: ~0.13.12 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.2.1
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

Setup: you can use this minimal project https://github.com/mccooll/reproduce-location-picker

1. Accept all location permissions.
2. Tap pick photo on Doogee device.
3. Take photo.
4. Hard close the app.
5. Re-open app.
6. Try taking a new photo and observe the failure.

### LINKED ISSUE #13665: launchImageLibraryAsync in expo-image-picker does not fire after app goes to the background in Android
### Summary

If I use ImagePicker once, then send the standalone app to the background by clicking the back button on Android until the app is closed, then open the app again, ImagePicker does not work. On debugging in a standalone app, I found that launchImageLibraryAsync did not fire.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

42

### Environment

Expo CLI 4.8.1 environment info:
    System:
      OS: Windows 10 10.0.22000
    Binaries:
      Node: 14.17.0 - C:\Program Files\nodejs\node.EXE
      npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  4.2.0.0 AI-202.7660.26.42.7351085
    npmPackages:
      expo: ^42.0.0 => 42.0.1
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

In Expo client, everything works as expected because the scenario below does not apply, please try the scenario below in a standalone app. I did not try in on iOS because I am on Windows.

Reproduction steps in a standalone app:
1. Open app
2. Click on button to pick image
3. Select image
4. Crop it
5. I get the image as expected

Then click the back button in Android multiple times until you get out of the app.
Repeat the steps above:
1. Open app
2. Click on button to pick image
3. Select image
4. BOOM... The picker is closed WITHOUT asking to crop the image and does not get any image because launchImageLibraryAsync did NOT fire.

If I kill the app and try again, everything works as expected.

### PR DESCRIPTION
**Pull request by mstach60161**
# Why

- Fixes #13665 
- Fixes #12102

# How

- Assigned correct `activity` after reopen app.

# Test Plan

Native component list.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt [hunk 1]
```diff
@@ -404,11 +404,15 @@ class ImagePickerModule(
   //region LifecycleEventListener
 
   override fun onHostDestroy() {
-    mWasDestroyed = true
-    mUIManager.unregisterLifecycleEventListener(this)
```
Comment (lukmccall) [inline packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt:408]: I think we should unregister the lifecycle listener in `onDestroy` (not it `onHostDestroy`) method.

**Review by lukmccall — APPROVED** [review]
LGTM 🥇 Just one thing

File: packages/expo-image-picker/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Fix promise resolving bug after putting app to background. ([#13975](https://github.com/expo/expo/pull/13975) by [@mstach60161](https://github.com/mstach60161))
```
Comment (bbarthec) [inline packages/expo-image-picker/CHANGELOG.md:13]:
```suggestion
- Fix promise not resolving when the app is moved to the background on Android. ([#13975](https://github.com/expo/expo/pull/13975) by [@mstach60161](https://github.com/mstach60161))
```

**Review by bbarthec — APPROVED** [review]
Looking good 👍 
Rephrased the CHANGELOG 😉

**Comment by tatocaster** [comment]
Hi. When this will be released in a regular release?

**Comment by bbarthec** [comment]
@tatocaster, I'm not sure what you mean by `a regular release`, but the new version of package containing all the unreleased features/bugs/fixes will be published to the npm in the 2 or 3 weeks (or possibly a bit sooner, but I can't promise that) and if you're asking about `managed workflow` (Expo Go with the new SDK version) it's going to happen in a month or so 👀

**Comment by tatocaster** [comment]
Thanks. This PR fixes the critical issue the team faces and I was curious about the dates.

**Pull request merged**